tests/unit/test_openstack.py

```diff
@@ -1150,3 +1389,63 @@ def test_pkgs_from_mixed_releases_found(self, mock_cli):
                "ussuri, victoria.")
         issues = list(IssuesStore().load().values())[0]
         self.assertEqual([issue['desc'] for issue in issues], [msg])
+
+
+class TestOpenstackApache2SSL(TestOpenstackBase):
+
+    def test_ssl_enabled(self):
+        with tempfile.TemporaryDirectory() as dtmp:
+            conf = "etc/apache2/sites-enabled/openstack_https_frontend.conf"
+            os.makedirs(os.path.dirname(os.path.join(dtmp, conf)))
+            with open(os.path.join(dtmp, conf), 'w') as fd:
+                fd.write(APACHE2_SSL_CONF)
+            setup_config(DATA_ROOT=dtmp)
+            base = openstack_core.OpenstackBase()
+            self.assertTrue(base.ssl_enabled)
+
+    def test_ssl_disabled(self):
+        base = openstack_core.OpenstackBase()
+        self.assertFalse(base.ssl_enabled)
+
+    def test_ssl_certificate_list(self):
+        with tempfile.TemporaryDirectory() as dtmp:
+            setup_config(DATA_ROOT=dtmp)
+            conf = "etc/apache2/sites-enabled/openstack_https_frontend.conf"
+            certificate_path = "etc/apache2/ssl/keystone/cert_10.5.100.2"
+            os.makedirs(os.path.dirname(os.path.join(dtmp, conf)))
+            os.makedirs(os.path.dirname(os.path.join(dtmp, certificate_path)))
+            with open(os.path.join(dtmp, conf), 'w') as fd:
+                fd.write(APACHE2_SSL_CONF)
+            with open(os.path.join(dtmp, certificate_path), 'w') as fd:
+                fd.write(CERTIFICATE_FILE)
+            base = openstack_core.OpenstackBase()
+            self.assertTrue(len(base.apache2_certificates_list) > 0)
+
+    def test_ssl_expiration_false(self):
+        # TODO we might need to change the assert here when the time comes :)
```

Comment:
We can't have a test whose state will change at some point in the future. I suggest that you just mock SSLCertificate.certificate_days_to_expire to return < 60 instead of relying on the contents of a real cert.